updater/updater.go

```diff
@@ -200,6 +251,144 @@ func upgradeIfRequired(checkURL string) (upgraded bool, err error) {
 	return true, nil
 }
 
+func getProfileFileName() (string, error) {
+	// File containing the profile info should exist with the updater binary.
+	updaterBin, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+	// Construct file name with absolute path.
+	profileFile := filepath.Join(filepath.Dir(updaterBin), profileFileName)
+	return profileFile, nil
+}
+
+func validateProfile(prf *Profile) error {
+	isAlphaNumeric := regexp.MustCompile(`^[A-Za-z0-9]+$`).MatchString
+	// Should validate the id string and channel.
+	// For now lets assume id string is less than 256
+	// characters and alphanumeric.
+	// Channel to be alphanemeric and less than 10 characters.
+	if !isAlphaNumeric(prf.Id) ||
+		!isAlphaNumeric(prf.Channel) ||
+		len(prf.Id) > 256 ||
+		len(prf.Channel) > 10 {
+		return errors.New("Profile Id or Channel format is invalid.")
+	}
+	return nil
+}
+
+func generateRandomIdString() (string, error) {
+	nRand, err := rand.Int(rand.Reader, big.NewInt(math.MaxInt64))
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s", nRand), nil
+}
+
+func loadProfile(prf *Profile) (err error) {
+	var fn string
+	var data []byte
+	if fn, err = getProfileFileName(); err != nil {
+		return err
+	}
+	if data, err = ioutil.ReadFile(fn); err != nil {
+		return err
+	}
+	if err = json.Unmarshal(data, prf); err != nil {
+		return err
+	}
+	return validateProfile(prf)
+}
+
+func saveProfile(prf *Profile) (err error) {
+	var fn string
+	var data []byte
+	if fn, err = getProfileFileName(); err != nil {
+		return err
+	}
+	if data, err = json.Marshal(prf); err != nil {
+		return err
+	}
+	return ioutil.WriteFile(fn, data, 0600)
+}
+
+func setRandomProfileId() int {
+	strRand, err := generateRandomIdString()
+	if err != nil {
+		fmt.Errorf("Error: %v\n", err)
```

Comment:
These `fmt.Errorf` are no-ops

Reply:
Will change these to fmt.Fprintf(os.Stderr, ...)